vpc.tf

```diff
@@ -1,48 +1,35 @@
+locals {
+  availability_zones = ["us-west-1a","us-west-1c"]
+}
+
 resource "aws_vpc" "main" {
-  cidr_block = "10.0.0.0/16"
+  cidr_block = var.vpc_cidr
 
   tags = {
     Name = var.env_code
   }
 }
 
-resource "aws_subnet" "public1" {
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.0.0/24"
-  availability_zone = "us-west-1a"
+resource "aws_subnet" "public" {
+  count = length(var.public_cidr)
 
-  tags = {
-    Name = "${var.env_code}-public1"
-  }
-}
-
-resource "aws_subnet" "public2" {
   vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.1.0/24"
-  availability_zone = "us-west-1c"
+  cidr_block        = var.public_cidr[count.index]
+  availability_zone = local.availability_zones[count.index]
 
   tags = {
-    Name = "${var.env_code}-public2"
+    Name = "${var.env_code}-public${count.index+1}"
   }
 }
 
-resource "aws_subnet" "private1" {
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.2.0/24"
-  availability_zone = "us-west-1a"
-
-  tags = {
-    Name = "${var.env_code}-private1"
-  }
-}
 
-resource "aws_subnet" "private2" {
+resource "aws_subnet" "private" {
   vpc_id            = aws_vpc.main.id
```

Comment:
could you please define the count variable and set it value.

Reply:
I have updated count variable and also fixed typo for subnet range in terraform.tfvars file. have pushed the changes in this branch to remote, please review.